src/cargo/util/progress.rs

```diff
@@ -76,6 +77,76 @@ struct Format {
     max_print: usize,
 }
 
+/// Taskbar progressbar
+///
+/// Outputs ANSI codes according to the `Operating system commands`.
+/// Currently only works in the Windows Terminal and ConEmu.
+struct TaskbarProgress {
+    show: bool,
+    pub percent: u8,
+}
+
+impl TaskbarProgress {
+    /// Creates a new `TaskbarProgress` from a cargo's config system.
+    ///
+    /// * `config == None` enables taskbar progress reporting on Windows
+    /// by default and disables on other platforms
+    fn from_config(config: Option<bool>) -> Self {
+        let show = match config {
+            Some(v) => v,
+            None => cfg!(target_os = "windows"),
```

Comment:
I don't think `target_os` is sufficient.  We should be checking the terminal as well.

Reply:
This is a blocker if we want to turn it on by default for certain platforms. Otherwise, we may figure it out later if it is an opt-in feature.

Reply:
Thank you for this link! For some reason I was thinking that this is much more complicated than reading env.
I've used a proper `ConEmuANSI` variable.
I haven't added OS check because we may run cargo in WSL and it "just works" now. 

Who-ho! This now works better then I expected.